### ISSUE TITLE
Unity support for Objectron?

### ISSUE THREAD

**Issue by ROBYER1** [google/mediapipe#1369]
Objectron looks absolutely fantastic, and to my understanding doesn't require ARCore(?), are there any plans to make a plugin for Unity? The use-cases on older phones that don't support ARCore and pairing with Unity 3d animation/code are limitless and I am very keen on trying this out in Unity. 

Unity forum post asking about this:
https://forum.unity.com/threads/ar-foundation-support-for-3d-object-detection-with-mediapipe.1016224/

**Comment by TigerHix** [comment]
Agreed—this can bring a _lot_ of people to use mediapipe in their VR/AR projects, many of which are built with Unity.

**Comment by ROBYER1** [comment]
> Agreed—this can bring a _lot_ of people to use MediaPipe in their VR/AR projects, many of which are built with Unity.

Good point, as a mixed reality developer, I wouldn't even dream of starting an AR project without Unity at this point. The platform has great support, getting started materials and asset management/optimisation in the engine for AR is second to none. Wasted a lot of time trying to craft AR  projects using other software/frameworks so would love to see MediaPipe in Unity for object detection + motion tracking at the very least!

**Comment by ROBYER1** [comment]
Please close if need be as this is a duplicate of https://github.com/google/mediapipe/issues/36

The use case for me is Instant Motion Tracking without the need for initialisation/ARCore/ARKit hardware requirements
- The user journey of wiggling the phone around for AR initialisation is not simple for some of our users to grasp (even still)
- Running of Objectron object detection would bypass some strict limitations of Image/object tracking used already in ARCore/ARKit - it is quite challenging to detect some objects by shape so we would prefer to train them using ML with Objectron or Object Detection YOLO style to track them with box detection or Objectron tracking

**Comment by midopooler** [comment]
Any conclusion to this thread?
I actually needed a human segmentation feature in unity (but for mobile applications)

**Comment by FerLuisxd** [comment]
+1, Having an API for Unity would be really nice to make 3D applications

**Comment by ROBYER1** [comment]
> @ROBYER1 BTW, you couldn't complete your task with https://github.com/homuler/MediaPipeUnityPlugin ?

It's nearly there, instant motion tracking support on IOS isn't working yet there sadly

**Comment by sgowroji** [comment]
Hi @ROBYER1, Could you please provide more information regarding the above issue if you still looking resolution for it. Thanks!

**Comment by Thaina** [comment]
We are still waiting for official unity package. Or the confirmation that google won't do it

**Comment by ROBYER1** [comment]
You could be waiting a while 😂😭

**Comment by FUBAR2077** [comment]
Is https://github.com/homuler/MediaPipeUnityPlugin not officially supported?

**Comment by Thaina** [comment]
> Is https://github.com/homuler/MediaPipeUnityPlugin not officially supported?

I think that was independent dev owned repo that not maintain by google isn't it?

**Comment by ilmarioranen** [comment]
I'd also REALLY love to see this supported in an official Unity repo / asset maintained by Google. It would mean MediaPipe is accessible to the biggest AR/XR/VR developer community on the planet and make a huge difference. 

Granted such a huge community might require additiobal attention and effort from Google, but why make this code public at all if the goal is not to make is accessible to as many people as possible?

**Comment by kotran88** [comment]
i'm now really into implement objectron on android device with s https://github.com/homuler/MediaPipeUnityPlugin repo but it's really hard to get there ...

**Comment by manwithsteelnerves** [comment]
We recently release [Easy ML Kit](https://u3d.as/2PMe) which targets to get google's ML Kit, Media Pipe and Tensorflow to Unity. Currently we released 2 features (Barcode scanning and object detection) and others are in developement.

We have **ARFoundation Camera** as one of the **input source**s which will be quite useful for AR Apps.
Would be great if you can share us what features you are expecting so we can prioritize.

**Comment by TZECHIN6** [comment]
> 

Hi @manwithsteelnerves , at this moment, seem that the custom object detection is not ready yet, and only 5 classes can be detected. 

Actually i am working on this [idea](https://github.com/google/mediapipe/issues/3436), the more I read in Google, the more I got confused...  Assume Unity is a must in my use case (for better graphic rending and flexibility of working on animation), which AI pipeline should I use for detecting the custom object and getting their corresponding location? 

For some Google library likes ARCore, ML Kit, Mediapipe, Easy ML Kit, TensorFlowLite... which framework should be used for my application?

**Comment by Thaina** [comment]
@TZECHIN6 Currently ARCore has its own session system to access hardware and fabricate many system together internally. It very rigid and I don't think it could integrate with mediapipe easily. Given that mediapipe itself also very rigid too

Given that ARCore require specific hardware to be able to work. I think mediapipe is more usable

**Comment by manwithsteelnerves** [comment]
@TZECHIN6 for all the features we developed(barcode, object detection, text recognition and face), we didn't notice any issues using ar camera as input for processing.
Regarding custom object detection model, thats something we will start on once we finish all features.

**Comment by moheydy** [comment]
After many years still you don't support of Unity 😐, I'm  sure 
The number of Unity developers who need mediapipe is 5 times that of Java and Android Studio ,objectron and body detection is really need for Ar and Xr Applications

**Comment by ys1317** [comment]
so does  anyone find a solution for that ( human body occlusion )

**Comment by Thaina** [comment]
It seem like the new mediapipe still not finish upgrade full body human yet

**Comment by akashvverma1995** [comment]
Hi @ROBYER1 

At the moment, We are not currently supporting Unity.
